library/core/src/Header.php

```diff
@@ -0,0 +1,240 @@
+<?php
+/**
+ * OpenEMR <http://open-emr.org>.
+ *
+ * @license https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Core;
+
+use Symfony\Component\Yaml\Yaml;
+use Symfony\Component\Yaml\Exception\ParseException;
+
+/**
+ * Class Header.
+ *
+ * Helper class to generate some `<script>` and `<link>` elements based on a
+ * configuration file. This file would be a good place to include other helpers
+ * for creating a `<head>` element, but for now it sufficently handles the
+ * `setupHeader()`
+ *
+ * @package OpenEMR
+ * @subpackage Core
+ * @author Robert Down <robertdown@live.com>
+ * @copyright Copyright (c) 2017 Robert Down
+ */
+class Header
+{
+
+    /**
+     * Setup various <head> elements.
+     *
+     * See root_dir/config/config.yaml for available assets
+     *
+     * Example usage in a PHP view script:
+     * ```php
+     * // Top of script with require_once statements
+     * use OpenEMR\Core\Header;
+     *
+     * // Inside of <head>
+     * // If no special assets are needed:
+     * Header::setupHeader();
+     *
+     * // If 1 special asset is needed:
+     * Header::setupHeader('key-of-asset');
+     *
+     * // If 2 or more assets are needed:
+     * Header::setupHeader(['array', 'of', 'keys']);
+     * ```
+     *
+     * Inside of a twig template (Parameters same as before):
+     * ```html
+     * {{ includeAsset() }}
+     * ```
+     *
+     * Inside of a smarty template, use | (pipe) delimited string of key names
+     * ```php
+     * {headerTemplate}
```

Comment:
for smarty, include the following example for when include optional assets(this is important since there are no examples of this yet in the codebase):
``````
{headerTemplate assets='key-of-asset'}  (1 optional assets)
{headerTemplate assets='array|of|keys'}  (multiple optional assets. ie. via | delimiter)
``````